linkerd/app/outbound/src/http/concrete.rs

```diff
@@ -19,6 +28,22 @@ pub struct ConcreteError {
     source: Error,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Endpoint<T> {
+    addr: Remote<ServerAddr>,
+    is_local: bool,
+    metadata: Metadata,
+    parent: T,
+}
+
+#[derive(Clone, Debug)]
+pub struct NewEndpoint<N> {
+    inner: N,
+    inbound_ips: IpSet,
+}
+
+pub type IpSet = Arc<AHashSet<IpAddr>>;
+
 impl<N> Outbound<N> {
     /// Builds a [`svc::NewService`] stack that builds buffered HTTP load
     /// balancer services for [`Concrete`] targets.
```

Comment:
comment should probably be updated since this no longer operates specifically on the `Concrete` type